### PR TITLE
MC-40651: A Configuration of a Configurable Product is reset when editing it in the Wish List

### LINKED ISSUE #32119: A Configuration of a Configurable Product is reset when editing it in the Wish List
**Preconditions:**
* Magento EE 2.4.1, EE 2.4.2, EE 2.4.2-p1 is installed
* A Customer account is created
* **A Configurable product is created:**
 name = TEST CONFIGURABLE 1;
 attribute = color;
 values = “red”, “black”;
 price of the “red” child product = $100;
 price of the “black” child product = $200.

**Steps to Reproduce:**
* Go to the Storefront as the created Customer
* Go to the created Product page
* Select “black <ins] $200.00” option in the “Color” drop-down
* Click the “ADD TO WISH LIST” button
* Make sure that:
 the price of the Product, added to the Wish List =$200
 hovering over “See Details” text trigger a pop-up displaying the selected option (Options Details
 Color black in this case)
* Click the “Edit item” button (“pencil” icon)

**(/) Expected results:**
* A Customer should be redirected to the editing Product page
* The displayed price should be =$200.00
* The option “black [/ins> $200.00” should be pre-selected in the “Color” drop-down

**Actual results:**

(/) A Customer is redirected to the editing Product page
 (x) The displayed price is =As low as $100.00
 (x) The option “black + $100.00” is not pre-selected in the “Color” drop-down

### PR DESCRIPTION
**Pull request by SmVladyslav**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#32119

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-32120`.

**Comment by SmVladyslav** [comment]
@magento run all tests

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontUpdateConfigurableProductAttributeOptionFromWishlistTest.xml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+            <!-- Assign Simple Products to Configurable Product -->
+            <createData entity="ConfigurableProductAddChild" stepKey="addConfigurableProductFirstChild">
+                <requiredEntity createDataKey="createConfigurableProduct"/>
+                <requiredEntity createDataKey="createFirstChildProduct"/>
+            </createData>
+            <createData entity="ConfigurableProductAddChild" stepKey="addConfigurableProductSecondChild">
+                <requiredEntity createDataKey="createConfigurableProduct"/>
+                <requiredEntity createDataKey="createSecondChildProduct"/>
+            </createData>
```
Comment (zakdma) [inline app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontUpdateConfigurableProductAttributeOptionFromWishlistTest.xml:81]: Please always perform
```
<magentoCron groups="index" stepKey="reindexInvalidatedIndices"/>
```
after new product attribute added

**Review by zakdma — CHANGES_REQUESTED** [review]

File: app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontUpdateConfigurableProductAttributeOptionFromWishlistTest.xml (hunk 1 above)
Comment (SmVladyslav) [reply app/code/Magento/Wishlist/Test/Mftf/Test/StorefrontUpdateConfigurableProductAttributeOptionFromWishlistTest.xml:81]: Fixed

**Comment by SmVladyslav** [comment]
@magento run all tests

**Comment by SmVladyslav** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Unit Tests

**Comment by SmVladyslav** [comment]
@magento run Functional Tests B2B

**Review by zakdma — APPROVED** [review]

**Comment by engcom-Oscar** [comment]
:heavy_check_mark: QA passed

Was checked case from Manual testing scenarios 
:x: **Before:**

(/) A Customer is redirected to the editing Product page
 (x) The displayed price is =As low as $100.00
 (x) The option “black + $100.00” is not pre-selected in the “Color” drop-down

:heavy_check_mark: **After:**

 A Customer redirected to the editing Product page
* The displayed price  = $200.00
* The option “black </ins> +100$ pre-selected in the “Color” drop-down
Screenshot from 2021-02-16 16-10-08

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-32120`.

**Pull request merged**